packages/fast-components-react-base/src/dialog/dialog.tsx

```diff
@@ -68,11 +102,24 @@ class Dialog extends Foundation<IDialogHandledProps & IManagedClasses<IDialogCla
         return (
             <div
                 className={get(this.props, "managedClasses.dialog_modalOverlay")}
+                onClick={this.handleOverlayClick}
                 role={"presentation"}
                 tabIndex={-1}
             />
         );
     }
+
+    private handleOverlayClick = (event: React.MouseEvent): void => {
+        if (this.props.onDismiss && this.props.visible) {
```

Comment:
you should also make sure `this.props.onDismiss` is a *function*, not just a truthy value. `typeof this.props.onDismiss === "function"`